subsys/bluetooth/mesh/adv_ext.c

```diff
@@ -321,6 +321,9 @@ void bt_mesh_adv_buf_relay_ready(void)
 			return;
 		}
 	}
+
+	/* Attempt to use the main adv set for the sending of relay messages. */
+	(void)schedule_send(&adv_main);
```

Comment:
Should it be something like this:
```suggestion
        if (CONFIG_BT_MESH_RELAY_ADV_SETS == 0) {
	    (void)schedule_send(adv);
	}
```
?

Or the whole function:
```
struct bt_mesh_ext_adv *adv = relay_adv_get();
int i = 0;

do {
    if (schedule_send(&adv[i])) {
        return;
    }
} while (++i < CONFIG_BT_MESH_RELAY_ADV_SETS);
```

Reply:
I might get this fix wrong. Will this actually make `adv_main` to pick up adv from relay queue?

Reply:
true